custom_components/sat/minimum_setpoint.py

```
@@ -1,87 +1,510 @@
+
+        # TRV-derived regime dimensions (coarse + hysteresis)
+        active_area_bucket = "sec_unknown"
+        demand_weight_bucket = "w_unknown"
+
+        if areas_snapshot is not None:
+            active_area_bucket = self._bucket_active_area_count_with_hysteresis(areas_snapshot.active_area_count)
+            demand_weight_bucket = self._bucket_demand_weight_with_hysteresis(areas_snapshot.demand_weight_sum or 0.0)
+
+        return f"{setpoint_band}:{temperature_band}:{load_band}:{active_area_bucket}:{demand_weight_bucket}"
+
+    def _bucket_active_area_count_with_hysteresis(self, active_area_count: int) -> str:
```

Pylint (4.0.4)

[refactor] Too many branches (16/12)

(R0912)